### ISSUE TITLE
All Colab notebooks stopped working today after spontaneous switch to A100

### ISSUE THREAD

**Issue by VisionaryMind** [googlecolab/colabtools#2925]
**Describe the current behavior**
Up until this afternoon, all pipelines have been working on the given V100 with 24GB. Today, Colab Pro+ switched us to an A100 16GB, and everything broke. We are getting the following error:

`ResizePyramid:	the provided PTX was compiled with an unsupported toolchain.
FilterH:	the provided PTX was compiled with an unsupported toolchain.
FilterV:	the provided PTX was compiled with an unsupported toolchain.`

**Describe the expected behavior**
There have been no changes to code. Custom routines are being compiled on the fly against the latest CUDA toolkit. If there is any potential problem here, it is the hard mandate for GCC 7 for compilation of a 3rd party package. That package also allows GCC 9, but compilation breaks with 9 in Colab.

**Comment by VisionaryMind** [comment]
Our notebooks download multiple GBs of proprietary data, which cannot be shared publicly. I have provided _you_ specifically with access to it and can include others, if you need. The fact of the matter is that this is a [well known problem](https://github.com/googlecolab/colabtools/issues/2452) that many of us have seen before and for which there was a fix applied. The problem arises from an incompatibility with the CUDA runtime when using certain GPUs.

Yes, we could dig up which CUDA toolkit version is compatible with V100, and spend several hours tinkering with that, but even if it were fixable, the larger issue is why are we now paying $40 / month for a service that offers us a mere V100 16GB? We need at least 24GB for our experiments, and V100 does not have noticeably better benchmark performance than our bare metal here at the lab. After upgrading to Pro+, we received an A100 24GB which was a bare minimum for our requirements. 

In other words, there is no sense in paying for Colab Pro+, as not only can you not change the type of GPU, additional GPUs cannot be added. So we, and at least ~50 of our colleagues that are using this service and seeing this problem, will likely be cancelling our subscription unless, of course, Google decides to allow us to choose what computing power our Colab instance will have. And for that, we would certainly be ready and willing to pay for. Historically, this hasn't been an option.

**Comment by VisionaryMind** [comment]
@ChiaraBi Welcome to the club. I have heard this story from at least 50 other colleagues. We are all searching for other alternatives. AWS is too expensive. Do you have any recommendations? I will post here should the situation change. While we are waiting, please run `nvidia-smi` on your Colab session and let me know what GPU / RAM size they have given you. For Pro+, they seem to be throttling limits down to the Pro or even Free tiers. Not sure what they are doing, but it's extraordinarily disruptive.

**Comment by VisionaryMind** [comment]
After a notebook refresh, it looks like Colab is now giving us a P100, still throttled at 16GB, but at least it allows us to run this particular pipeline. Granted, it's much slower than the RTX 3080 on my desktop, but it is (technically) running. Other pipelines requiring minimum 24GB GPU RAM will fail, however. Some even require 40GB. And it's a mystery to me why higher-powered GPUs in the Colab infrastructure run far slower than on bare metal. I put the Colab P100 and RTX side by side, and RTX ended up being almost 2x faster. So consider this when paying for the service.

@ChiaraBi, we are now looking at [Paperspace Gradient](https://gradient.run/). In addition to not having these restrictions, at a similar pricepoint, there is also persistent storage, so no more waiting on Colab for 30 minutes while it re-installs dependencies between broken sessions. I will give it a try and report back. My colleagues are also trying other options before tabling Colab, namely Kaggle and Sagemaker.

So you can roll the dice and hope Google gives you a better (workable) GPU tomorrow or just graduate to non-hobbyist infrastructure. In a sense, it's a good thing this happened, or else we may not have known what we were missing.

**Comment by ChiaraBi** [comment]
@VisionaryMind this is what I got:
<img width="678" alt="Screenshot 2022-07-15 at 17 18 02" src="https://user-images.githubusercontent.com/10097737/179254871-91c6cf72-bbd9-42b4-9f8d-8021f75cbe07.png">

I was thinking of trying Kaggle as well, thanks for the suggestions!

**Comment by VisionaryMind** [comment]
Your GPU / RAM are expected for Pro tier. Have you enabled High-RAM? If, then 16GB is too low. We have enabled high-RAM for Pro+, and we don't get anything higher than 16GB. For the error you are seeing, it's coming from Torch. You may want to either upgrading to a newer version of Torch or modifying your CUDA toolkit installation, although -- I don't believe the latter is needed with Torch installed. We use Miniconda environments to handle these dependencies. I have found it impossible to compile CUDA code without moving to an isolated Conda environment and installing everything under strict version control.

**Comment by VisionaryMind** [comment]
UPDATE: We had a very important pipeline running for 600 epochs overnight. Just ~30 minutes ago, after reaching Epoch 597, Google spontaneously shut down the runtime. It was not an OOM issue or some other issue with the script, as the logs showed no sign of issue. This can only mean one thing -- that even for Colab Pro+ customers, the infrastructure is not stable and cannot be trusted to critical jobs. @ChiaraBi, we ran tests with Paperspace Gradient yesterday, and it performed quite well, but the free-tier A100 GPUs have a 6-hour time-limit. So we have to make a decision --- work around Gradient's time limits for better stability or roll the die with Google. I am leaning toward the latter. Will look forward to hearing other suggestions from anyone interested in commenting.

**Comment by VisionaryMind** [comment]
We have officially moved off Google Colab and have asked for a refund. The Google team is not only in violation of the terms of Pro+, according to our understanding, by not providing higher tear GPUs with more memory than Pro or free services, but also does not appear to understand the nuances of the services it is providing. Just now, code that was compiling and running for over 2 months now stopped compiling with absolutely no usable error message. I will avoid posting any further details here.

If there is anyone else out there that is seeing these problems and has paid 1 or months for Google Colab Pro+ service, please reach out to me. I am collecting names.

**Comment by SkelegonDK** [comment]
I have the same issue, never had issues before until now.
